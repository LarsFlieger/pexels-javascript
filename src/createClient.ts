require("isomorphic-fetch");

import generatePhotoEndPoints from "./generatePhotoEndpoints";

export default function createClient(apiKey: string) {
  if (!apiKey || typeof apiKey !== "string") {
    throw new TypeError(
      `An ApiKey must be provided when initiating the Pexel's client.`
    );
  }

  return {
    photos: generatePhotoEndPoints(apiKey),
  };
}
