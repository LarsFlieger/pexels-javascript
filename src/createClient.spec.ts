import createClient from "./createClient";

describe("#createClient", () => {
  test("smoke test", () => {
    expect(() => createClient("api key")).not.toThrowError();
  });

  test("throws an error if an apiKey is not provided", () => {
    expect(() => createClient("")).toThrowError();
  });
});
