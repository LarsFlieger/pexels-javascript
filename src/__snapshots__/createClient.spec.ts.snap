// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#createClient snapshots client.photos.curated 1`] = `
Object {
  "next_page": null,
  "page": 1,
  "per_page": 15,
  "photos": Array [
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": null,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": null,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
  ],
}
`;

exports[`#createClient snapshots client.photos.curated 2`] = `
Object {
  "next_page": null,
  "page": 1,
  "per_page": 10,
  "photos": Array [
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": null,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
  ],
}
`;

exports[`#createClient snapshots client.photos.search 1`] = `
Object {
  "next_page": null,
  "page": 1,
  "per_page": 15,
  "photos": Array [
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
  ],
  "total_results": null,
}
`;

exports[`#createClient snapshots client.photos.search 2`] = `
Object {
  "next_page": null,
  "page": 1,
  "per_page": 4,
  "photos": Array [
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
    Object {
      "height": null,
      "id": null,
      "liked": false,
      "photographer": null,
      "photographer_id": null,
      "photographer_url": null,
      "src": Object {
        "landscape": null,
        "large": null,
        "large2x": null,
        "medium": null,
        "original": null,
        "portrait": null,
        "small": null,
        "tiny": null,
      },
      "url": null,
      "width": null,
    },
  ],
  "total_results": null,
}
`;
